### PR TITLE
[moveit_cpp] Add setLastPlanSolution

### PR DESCRIPTION
**Pull request by galou**
Signed-off-by: Gaël Écorchard <gael.ecorchard@cvut.cz>

### Description

Without this function, it is not possible to execute another plan that the last computed one. I don't like the name (but I don't like the idea of having `execute()`  not taking a plan as input either), maybe `setNextPlanToExecute()` could be an alternative. So, it's more to start a discussion.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h [hunk 1]
```diff
@@ -177,9 +177,8 @@ class MoveItCpp
   planning_scene_monitor::PlanningSceneMonitorPtr planning_scene_monitor_;
 
   // Planning
-  std::map<std::string, planning_pipeline::PlanningPipelinePtr> planning_pipelines_;
-  std::map<std::string, std::set<std::string>> groups_pipelines_map_;
-  std::map<std::string, std::set<std::string>> groups_algorithms_map_;
```
Comment (henningkayser) [inline moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h:182]: Is this change related in any way?

**Review by henningkayser — COMMENTED** [review]
@galou I'm not convinced that this addition is necessary. You can execute arbitrary trajectories using `MoveItCpp::execute()`. In fact, you can even access the `TrajectoryExecutionManager` and mess with it during execution. The `PlanningComponent` is meant to be a convenience wrapper for planning requests and not really an execution interface. I would rather remove the `PlanningComponent::execute()`, probably. 

Side note: `getLastPlanSolution()` returns a pointer to non-const (which is not really ideal). That would also allow you to change the last solution trajectory.

File: moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h (hunk 1 above)
Comment (galou) [reply moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h:182]: Thanks, `MoveItCpp::execute()` is the answer I was looking for.

Wouldn't it be also better to remove `getLastPlanSolution()`? I don't see the use case.

Comment (henningkayser) [reply moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h:182]: I think being able to introspect or process the solution makes sense, but maybe it should really be non-const.
